nowcasting_dataset/dataset/batch.py

```diff
@@ -183,3 +229,63 @@ def data_sources(self):
             self.gsp,
             self.nwp,
         ]
+
+
+def join_two_batches(
+    batches: List[Batch],
+    data_sources_names: Optional[List[str]] = None,
+    first_batch_examples: Optional[List[int]] = None,
+) -> Batch:
+    """
+    Join two batches
+
+    Args:
+        batches: list of batches to be mixes
+        data_sources_names: list of data source names
+        first_batch_examples: lsit of indexes that we should use for the first batch
+
+    Returns: batch object, mixture of two given
+
+    """
+
+    if len(batches) == 1:
+        return batches[0]
+
+    assert len(batches) == 2, f"Can only join list of two batches, was given {len(batches)} batches"
+
+    if data_sources_names is None:
+        data_sources_names = list(Example.__fields__.keys())
+
+    batch = batches[0]
+    batch_size = batch.metadata.batch_size
+
+    # create random indexes, if needed
+    if first_batch_examples is None:
+        first_batch_examples = np.random.randint(low=0, high=batch_size, size=int(batch_size / 2))
+    second_batch_examples = [i for i in range(0, batch_size) if i not in first_batch_examples]
```

Comment:
If we want them to be random, why not have random examples from both? As in, as it is here, if the 0th and 2nd examples are chosen from the first batch, they can't be chosen from the second. For it to be more random, it could allow for having the 0th and 2nd examples of the first and second batch in one batch together, if that makes sense! I get it might make it a bit more complicated as we then might not be able to concat along the example?

Reply:
That is a very good point. That would make it more random, but the code would be a little be more complicated as you said. 

ill have a think whats the best thing to do for the long-term

Reply:
I went with your idea - thanks @jacobbieker - makes it all a bit more random